### PR TITLE
New source code extraction for C++

### PR DESCRIPTION
**Pull request by ethanbwang**

### CODE REVIEW

**Comment by ethanbwang** [comment]
I think for Hongxiang's PR we forgot to update init.sh to install the new packages in `source_code.py`.

**Comment by harrison4anewbe** [comment]
Which PR?

**Comment by ethanbwang** [comment]
https://github.com/SecurityLab-UCD/OSS-Fuzz-scripts/pull/33
I think you imported new libraries in `source_code.py`

File: scripts/source_code.py [hunk 1]
```diff
@@ -13,7 +13,10 @@
 def clang_get_func_code(
     file_path: str,
     function_name: str,
-    get_node_name: Callable[[clang.cindex.Cursor], str] = lambda node: node.spelling,
+    get_node_name: Callable[
```
Comment (EYH0602) [inline scripts/source_code.py:16]: Please use a more descriptive name for this parameter

File: scripts/source_code.py [hunk 2]
```diff
@@ -13,7 +13,10 @@
...
+        [clang.cindex.Cursor], str
+    ] = lambda node, name: node.spelling
+    == name,
```
Comment (EYH0602) [inline scripts/source_code.py:19]: The type of this parameter is wrong. Please check https://docs.python.org/3/library/typing.html#annotating-callable-objects

**Review by EYH0602 — CHANGES_REQUESTED** [review]

File: scripts/source_code.py [hunk 3]
```diff
@@ -13,7 +13,10 @@
 def clang_get_func_code(
     file_path: str,
     function_name: str,
-    get_node_name: Callable[[clang.cindex.Cursor], str] = lambda node: node.spelling,
+    correct_node: Callable[
```
Comment (EYH0602) [inline scripts/source_code.py:16]: `is_correct_node` might be better

**Review by EYH0602 — APPROVED** [review]

**Pull request merged**